presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/rtree/TestFlatbush.java

```diff
@@ -0,0 +1,281 @@
+        OGCGeometryWrapper pointY = new OGCGeometryWrapper(POINT_Y);
+        OGCGeometryWrapper pointZ = new OGCGeometryWrapper(POINT_Z);
+        OGCGeometryWrapper pointW = new OGCGeometryWrapper(POINT_W);
+
+        Flatbush rtree = new Flatbush<OGCGeometryWrapper>(new OGCGeometryWrapper[] {
+                pointX, pointY, pointZ, pointW
+        });
+
+        List<OGCGeometryWrapper> resultsA = rtree.findIntersections(octagonA.getExtent());
+        resultsA.sort(Comparator.naturalOrder());
+        assertEquals(resultsA, Arrays.asList(pointX, pointY));
+
+        List<OGCGeometryWrapper> resultsB = rtree.findIntersections(octagonB.getExtent());
+        resultsB.sort(Comparator.naturalOrder());
+        assertEquals(resultsB, Arrays.asList(pointY, pointZ));
+    }
+
+    @Test
+    public void testOctagonTree()
+    {
+        OGCGeometryWrapper octagonA = new OGCGeometryWrapper(POLYGON_A);
+        OGCGeometryWrapper octagonB = new OGCGeometryWrapper(POLYGON_B);
+
+        OGCGeometryWrapper pointX = new OGCGeometryWrapper(POINT_X);
+        OGCGeometryWrapper pointY = new OGCGeometryWrapper(POINT_Y);
+        OGCGeometryWrapper pointZ = new OGCGeometryWrapper(POINT_Z);
+        OGCGeometryWrapper pointW = new OGCGeometryWrapper(POINT_W);
+
+        Flatbush rtree = new Flatbush<OGCGeometryWrapper>(new OGCGeometryWrapper[] {octagonA, octagonB});
+
+        List<OGCGeometryWrapper> resultsX = rtree.findIntersections(pointX.getExtent());
+        assertEquals(resultsX, Arrays.asList(octagonA));
+
+        List<OGCGeometryWrapper> resultsY = rtree.findIntersections(pointY.getExtent());
+        resultsY.sort(Comparator.naturalOrder());
+        assertEquals(resultsY, Arrays.asList(octagonA, octagonB));
+
+        List<OGCGeometryWrapper> resultsZ = rtree.findIntersections(pointZ.getExtent());
+        assertEquals(resultsZ, Arrays.asList(octagonB));
+
+        List<OGCGeometryWrapper> resultsW = rtree.findIntersections(pointW.getExtent());
+        assertTrue(resultsW.isEmpty());
+    }
+
+    private void checkRandomRects(int numBuildRects, int numProbeRects, int seed)
+    {
+        Random rng = new Random(seed);
+        List<Rectangle> buildRects = TestUtils.makeRandomRectangles(rng, numBuildRects);
+        List<Rectangle> probeRects = TestUtils.makeRandomRectangles(rng, numProbeRects);
+
+        Flatbush rtree = new Flatbush<Rectangle>(buildRects.toArray(new Rectangle[] {}));
+        for (Rectangle query : probeRects) {
+            List rtreeResults = rtree.findIntersections(query);
+            rtreeResults.sort(rectangleComparator);
+            List bruteResults = buildRects.stream()
+                    .filter(rect -> rect.intersects(query))
+                    .collect(Collectors.toList());
+            bruteResults.sort(rectangleComparator);
+            assertEquals(rtreeResults, bruteResults);
+        }
+    }
+
+    @Test
+    public void testRandomRects100Probe1000Build()
```

Comment:
- Don't abbreviate.
- I think a better structure would to be have helper method take two lists of rectangles and have the tests compose these lists as they see fit.

```
    @Test
    public void testRandomRectanglesSmall()
    {
        Random random = new Random(42);
        assertRectangles(makeRectangles(random, 1_000), makeRectangles(random, 100));
    }

    @Test
    public void testRandomRectanglesMedium()
    {
        Random random = new Random(123);
        assertRectangles(makeRectangles(random, 10_000), makeRectangles(random, 1_000));
    }

    @Test
    public void testRandomRectanglesLarge()
    {
        Random random = new Random(321);
        assertRectangles(makeRectangles(random, 50_000), makeRectangles(random, 5_000));
    }

    private static void assertRectangles(List<Rectangle> buildRectanges, List<Rectangle> probeRectanges)
    {
        Flatbush<Rectangle> rtree = new Flatbush<>(buildRectanges.toArray(new Rectangle[] {}));
        for (Rectangle probe : probeRectanges) {
            List<Rectangle> expected = buildRectanges.stream()
                    .filter(rectangle -> rectangle.intersects(probe))
                    .collect(toImmutableList());
            assertEqualsNoOrder(findIntersections(rtree, probe), expected);
        }
    }

    private static void assertEqualsNoOrder(List<Rectangle> actual, List<Rectangle> expected)
    {
        List<Rectangle> actualSorted = actual.stream().sorted(RECTANGLE_COMPARATOR).collect(toImmutableList());
        List<Rectangle> expectedSorted = expected.stream().sorted(RECTANGLE_COMPARATOR).collect(toImmutableList());

        assertEquals(actualSorted, expectedSorted);
    }
```